standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -870,44 +870,88 @@ private boolean isViewTable(String catName, String dbName, String tblName) throw
+    // sets s limit on the number of parameters that can be set to 32767. This is also the number of parameters
+    // that can be passed to the JDO executeArray call.
+    final int JDBC_STMT_PARAM_LIMIT = 32767;
+
     final String dbNameLcase = dbName.toLowerCase();
     final String tblNameLcase = tblName.toLowerCase();
     final String catNameLcase = normalizeSpace(catName).toLowerCase();
 
     // We have to be mindful of order during filtering if we are not returning all partitions.
+    // If the number of parameters is larger than 32767 and the query is split into multiple
+    // parts, the order by would work within each query part.
     String orderForFilter = (max != null) ? " order by \"PART_NAME\" asc" : "";
 
     String queryText =
-        "select " + PARTITIONS + ".\"PART_ID\" from " + PARTITIONS + ""
-      + "  inner join " + TBLS + " on " + PARTITIONS + ".\"TBL_ID\" = " + TBLS + ".\"TBL_ID\" "
-      + "    and " + TBLS + ".\"TBL_NAME\" = ? "
-      + "  inner join " + DBS + " on " + TBLS + ".\"DB_ID\" = " + DBS + ".\"DB_ID\" "
-      + "     and " + DBS + ".\"NAME\" = ? "
-      + join(joinsForFilter, ' ')
-      + " where " + DBS + ".\"CTLG_NAME\" = ? "
-      + (StringUtils.isBlank(sqlFilter) ? "" : (" and " + sqlFilter)) + orderForFilter;
-    Object[] params = new Object[paramsForFilter.size() + 3];
-    params[0] = tblNameLcase;
-    params[1] = dbNameLcase;
-    params[2] = catNameLcase;
-    for (int i = 0; i < paramsForFilter.size(); ++i) {
-      params[i + 3] = paramsForFilter.get(i);
-    }
+            "select " + PARTITIONS + ".\"PART_ID\" from " + PARTITIONS + ""
+                    + "  inner join " + TBLS + " on " + PARTITIONS + ".\"TBL_ID\" = " + TBLS + ".\"TBL_ID\" "
+                    + "    and " + TBLS + ".\"TBL_NAME\" = ? "
+                    + "  inner join " + DBS + " on " + TBLS + ".\"DB_ID\" = " + DBS + ".\"DB_ID\" "
+                    + "     and " + DBS + ".\"NAME\" = ? "
+                    + join(joinsForFilter, ' ')
+                    + " where " + DBS + ".\"CTLG_NAME\" = ? "
+                    + (StringUtils.isBlank(sqlFilter) ? "" : (" and " + sqlFilter)) + orderForFilter;
 
-    long start = doTrace ? System.nanoTime() : 0;
-    Query query = pm.newQuery("javax.jdo.query.SQL", queryText);
-    List<Object> sqlResult = executeWithArray(query, params, queryText, ((max == null)  ? -1 : max.intValue()));
-    long queryTime = doTrace ? System.nanoTime() : 0;
-    MetastoreDirectSqlUtils.timingTrace(doTrace, queryText, start, queryTime);
-    if (sqlResult.isEmpty()) {
-      return Collections.emptyList(); // no partitions, bail early.
+    int parametersToProcess = paramsForFilter.size();
+
+    List<Long> result = new ArrayList<>();
+
+    // If this method has been called by default we need to query the tables once. Hence set the minimum numbner of
+    // iterations to 1.
+    int iterations = 1;
+
```

Comment:
remove extra blank lines. max of 1 blank line between code blocks.

Reply:
Will do ! Will make the change and resubmit.